shark_turbine/kernel/wave/constraints.py

```diff
@@ -92,7 +103,27 @@ def subs_vector_shapes(self, index_map: dict[IndexSymbol, int]):
             if isinstance(vector_size, IndexExpr):
                 self.vector_shapes[vector_dim] = vector_size.subs(index_map)
 
-    def apply(self, mma_index: int) -> IndexSequence:
+    def compute_access_pattern_using_vector_shapes(
+        self,
+        dim: IndexSymbol,
+        workgroup_dim: int,
+        elements_per_thread: int | IndexSymbol,
+    ) -> IndexSequence:
+        if dim not in self.vector_shapes:
+            raise ValueError(f"No vector shape specified for dimension {dim}")
+        thread_id = self.get_thread_id_from_workgroup_dim(workgroup_dim)
+        return IndexSequence(thread_id * elements_per_thread, elements_per_thread, 1)
+
+    def apply(
+        self,
+        mma_index: int,
+        dim: IndexSymbol,
+        elements_per_thread: int | IndexSymbol,
+    ) -> IndexSequence:
+        if self.vector_shapes is not None:
+            return self.compute_access_pattern_using_vector_shapes(
```

Comment:
For matmul applications, would it ever go to the Piecewise function part? This looks like we would always return early, since I reckon even for MMA cases we'd have `self.vector_shapes` ?